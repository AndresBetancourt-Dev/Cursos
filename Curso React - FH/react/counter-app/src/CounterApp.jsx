import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = (e) => {
    setCounter((c) => c + 1);
  };

  const handleSubstact = (e) => {
    setCounter(counter - 1);
  };

  const handleReset = (e) => {
    setCounter(value);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{value}</h2>
      <h3>{counter}</h3>
      <button className={"button counter"} onClick={handleAdd}>
        +1
      </button>
      <button className={"button counter"} onClick={handleReset}>
        Reset
      </button>
      <button className={"button counter"} onClick={handleSubstact}>
        -1
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

export { CounterApp };
