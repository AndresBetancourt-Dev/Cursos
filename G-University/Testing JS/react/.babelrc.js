module.exports = {
  presets: [
    ['@babel/preset-env', {targets: {node: 'current'}}],
    [
      "@babel/preset-react",
      {
        development: process.env.BABEL_ENV === "development",
      },
    ]
  ],
};
