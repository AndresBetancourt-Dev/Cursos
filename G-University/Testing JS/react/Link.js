import React from "react";

const STATUS = {
  HOVERED: "hovered",
  NORMAL: "normal",
};

const Link = ({ page, children }) => {
  const [status, setStatus] = React.useState(STATUS.NORMAL);

  const onMouseEnter = () => {
    setStatus(STATUS.HOVERED);
  };

  const onMouseLeave = () => {
    setStatus(STATUS.NORMAL);
  };

  return (
    <a
      className={status}
      href={page || "#"}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      {children}
    </a>
  );
};

export default Link;
