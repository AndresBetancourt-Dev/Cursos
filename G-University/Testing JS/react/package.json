{
  "name": "react",
  "version": "1.0.0",
  "description": "",
  "main": "Link.js",
  "dependencies": {
    "react": "^18.2.0",
    "react-test-renderer": "^18.2.0"
  },
  "devDependencies": {
    "@babel/core": "^7.20.5",
    "@babel/preset-env": "^7.20.2",
    "@babel/preset-react": "^7.18.6",
    "babel": "^6.23.0",
    "babel-cli": "^6.26.0",
    "babel-jest": "^29.3.1",
    "jest": "^29.3.1"
  },
  "scripts": {
    "test": "jest",
    "test:update": "jest --updateSnapshot --watch"
  },
  "author": "",
  "license": "ISC"
}
