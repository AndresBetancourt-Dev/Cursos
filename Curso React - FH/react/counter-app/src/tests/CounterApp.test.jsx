import { describe, test, expect, beforeEach } from "@jest/globals";
import React from "react";
import { shallow } from "enzyme";
import { CounterApp } from "../CounterApp";
import { render } from "@testing-library/react";

describe("Test for the Counter App", () => {
  let value = 0;
  let Component = shallow(<CounterApp value={value} />);

  beforeEach(() => {
    value = 0;
    Component = shallow(<CounterApp value={value} />);
  });

  test("Match Snapshot", () => {
    expect(Component).toMatchSnapshot();
  });

  test("Match Value sent by Props", () => {
    const Component = shallow(<CounterApp value={value} />);

    const renderValue = Component.find("h2").text().trim();

    expect(renderValue).toBe(value.toString());
    expect(renderValue).not.toBe("101");
  });

  test("Add Button Click Test", () => {
    Component.find("button").at(0).simulate("click");
    const renderValue = Component.find("h3").text().trim();

    expect(renderValue).toEqual("1".toString());
  });

  test("Substract Button Click Test", () => {
    Component.find("button").at(2).simulate("click");
    const renderValue = Component.find("h3").text().trim();

    expect(renderValue).toEqual((value - 1).toString());
  });

  test("Reset Button Click Test", () => {
    Component.find("button").at(0).simulate("click");
    Component.find("button").at(0).simulate("click");
    Component.find("button").at(0).simulate("click");
    Component.find("button").at(0).simulate("click");
    Component.find("button").at(0).simulate("click");
    Component.find("button").at(0).simulate("click");
    Component.find("button").at(0).simulate("click");
    Component.find("button").at(0).simulate("click");
    Component.find("button").at(0).simulate("click");
    Component.find("button").at(0).simulate("click");

    let renderValue = Component.find("h3").text().trim();

    expect(renderValue).toBe("10");

    Component.find("button").at(1).simulate("click");

    renderValue = Component.find("h3").text().trim();

    expect(renderValue).toBe(value.toString());
    expect(renderValue).not.toBe("10");
  });
});
