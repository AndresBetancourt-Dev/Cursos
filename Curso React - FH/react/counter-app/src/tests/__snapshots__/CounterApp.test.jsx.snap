// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the Counter App Match Snapshot 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <h3>
    0
  </h3>
  <button
    className="button counter"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="button counter"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    className="button counter"
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
